src/roomba_robot/botgal_and_lib/main.c

```diff
@@ -0,0 +1,196 @@
+#include <stdbool.h>
+#include "kipr/wombat.h"
+
+void moveservo(int port, int position) {
+    enable_servos();
+    set_servo_position(port, position);
+    msleep(1000);
+}
+
+void movemotor(int port, int position, int motorpower, int limitswitchport) {
+    clear_motor_position_counter(port);
+    if (digital(limitswitchport) == 0) {
```

Comment:
Invert this if-statement to remove nesting by one level:

```c
void movemotor(int port, int position, int motorpower, int limitswitchport) {
    clear_motor_position_counter(port);
    if (digital(limitswitchport) != 0) return;
    
    while (get_motor_position_counter(port) <= position) {
        motor(port, motorpower);
        msleep(10);
    }
}
```

Reply:
in the current testing state it makes sense for us to have two different main.c files because the code is spread between multiple different projects. When we finish with it we will combine it into one